src/FileMigrationHelper.php

```diff
@@ -102,6 +102,17 @@ protected function migrateFile($base, File $file, $legacyFilename)
             return false;
         }
 
+        // If this file has the wrong class for it's extension, update the classname
+        $destinationClass = $file->get_class_for_file_extension($extension);
+        if ($file->ClassName !== $destinationClass) {
+            // We disable validation so that we are able to write a corrected
+            // classname, once that is changed we re-enable it for subsequent writes
+            Config::modify()->set(File::class, 'validation_enabled', false);
+            $file = $file->newClassInstance($destinationClass);
+            $file->write();
+            Config::modify()->set(File::class, 'validation_enabled', true);
```

Comment:
I _think_ that doing this will keep a nested config manifest around, whereas if you explicitly nest it then change the config prop, then unnest it it will remove the nested manifest for the rest of the execution. Thoughts?

Reply:
I tried nesting it, but it doesn't work when I modify it that way.